nntrainer/src/neuralnet.cpp

```diff
@@ -499,29 +449,19 @@ int NeuralNetwork::init() {
     const char *acti_str =
       iniparser_getstring(ini, (layer_name + ":Activation").c_str(), unknown);
     ActiType act = (ActiType)parseType(acti_str, TOKEN_ACTI);
-
     layers.back()->setActivation(act);
-    status = initActivationLayer(act);
-    NN_INI_RETURN_STATUS();
 
     /** Add flatten layer */
     bool flatten =
       iniparser_getboolean(ini, (layer_name + ":Flatten").c_str(), false);
-    if (flatten) {
-      status = initFlattenLayer();
-      NN_INI_RETURN_STATUS();
-    }
+    layers.back()->setFlatten(flatten);
+
     previous_dim = layers.back()->getOutputDimension();
   }
```

Comment:
I'm not sure that this has meaning. output dimension is set when the layer is initialized with input dimension. If you are going to initialize layer later, then this could be removed.

Reply:
Yes, removed.